torch/csrc/jit/runtime/symbolic_script.cpp

```diff
@@ -1446,6 +1446,12 @@ const std::vector<std::string> functions = {
             return grad_output * mask, None
           return torch.hardshrink(self, lambd=lambd), backward
 
+        def hardtanh(self, min_val: number, max_val: number):
+          def backward(grad_output):
+            mask = ((self >= float(min_val)) * (self <= float(max_val))).type_as(self)
```

Comment:
Can you use `&` here instead of `*` ? that is more inline with what we are doing. I dont think you need the `type_as` call. 

Why do you call `float(min_val)` ? that might be throwing away precision for integer inputs.